cpp/examples/tutorial_examples/arrow_example.cc

```diff
@@ -0,0 +1,87 @@
+#include <arrow/api.h>
+
+#include <iostream>
+
+arrow::Status RunMain(int argc, char** argv) {
```

Comment:
Instead of naming all functions as `RunMain`, I recommend to use a representative name as if it was part of "real" code. Also, maybe some of the `RunMain` could be split into multiple functions, like a: `MakeArrays()`, `MakeTableFromChunks()`. But not sure how the return value would look like for a collection of items. Arrow has `Result` which can be a value or a `Status`, but I do not think it handles sequences.

Reply:
I assume that if we put it all in a vector, we can return a Result object and go from there. I'll try that out soon. 

However, for this, I'm not certain we actually want to make sure it resembles real code -- this is for on-boarding, rather than cookbooks or anything like that. I think splitting it up removes some of the linear feeling from the code, and I'd rather make it so this is very precise to read -- it shouldn't be scrolling between methods, but looking at what was done directly above the current code. 

However, that may just be me, and I'm open to changing it.

Reply:
Ok, I am good with continuing to use `RunMain`.